### PR TITLE
Remove header rom filter

### PR DESCRIPTION
**Pull request by dima-xd**
People were saying it's useless. Not sure if I did it correctly

### CODE REVIEW

**Comment by AbandonedCart** [comment]
Isn't that what separates formats when you have more than one?

**Comment by dima-xd** [comment]
> Isn't that what separates formats when you have more than one?



> Isn't that what separates formats when you have more than one?

it's just a filter among all roms (thing under search)

**Comment by AbandonedCart** [comment]
> > Isn't that what separates formats when you have more than one?
> 
> it's just a filter among all roms (thing under search)

Yes. It separates your files by the format / extension (ie. NSP, XCI). What you're removing here is the heading that goes between sections. I would wager the people calling it useless only have files with a single extension.

**Comment by dima-xd** [comment]
> > it's just a filter among all roms (thing under search)
> 
> Yes. It separates your files by the format / extension (ie. NSP, XCI). What you're removing here is the heading that goes between sections. I would wager the people calling it useless only have files with a single extension.

devs said it's useless too :)

**Comment by AbandonedCart** [comment]
Probably deemed it unnecessary, but that's still the actual function of it.

File: app/src/main/java/emu/skyline/MainActivity.kt [hunk 1]
```diff
@@ -304,14 +294,18 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun populateAdapter() {
+        binding.textTitle.text = null
+        binding.textTitle.visibility = View.GONE
         val items = getDataItems()
         adapter.setItems(items.map {
             when (it) {
```
Comment (AbandonedCart) [inline app/src/main/java/emu/skyline/MainActivity.kt:301]: Couldn't you drop most of this? The whole reason for the map and when was that there could be more than one type. That's no longer the case.

**Comment by AbandonedCart** [comment]
You have a `RomFile` that spawns an `AppEntry` that spawns a `DataItem` of the `AppItem` type because some `AppItem`s could be `HeaderItem`s. If it can no longer be a `HeaderItem`, is `AppItem` still a necessary assignment?

**Comment by lynxnb** [comment]
We definitely want to get rid of all the unnecessary filtering code too. If either of you @AbandonedCart or @dima-xd want to do that either in this PR or as a follow-up one, we'd appreciate it very much. Otherwise, as always, I'll get to this at some point.

File: app/src/main/res/layout/main_activity.xml [hunk 2]
```diff
@@ -113,6 +113,16 @@
                     app:cardUseCompatPadding="true"
                     app:layout_constraintTop_toBottomOf="@id/sub_text"
                     app:shapeAppearance="@style/ShapeAppearance.Material3.Corner.ExtraLarge" />
+
+                <TextView
+                    android:id="@+id/text_title"
```
Comment (lynxnb) [inline app/src/main/res/layout/main_activity.xml:118]: Give a more meaninful name to this text view, since it's only used with the "No ROMs found" text.

**Review by lynxnb — COMMENTED** [review]
IMO the game list needs a bit of padding at the top from the search bar, currently it's too clamped up.

**Comment by dima-xd** [comment]
> > We definitely want to get rid of all the unnecessary filtering code too.
> 
> I have a version of it that respects the settings option to sort by extension, but removes the overhead of the secondary array. That would be the least destructive version. I didn't go any further without knowing if the sorting option is staying. [8bitDream@8b9febd](https://github.com/8bitDream/skyline/commit/8b9febd6700da7ecac8bf3b90a5f6fe1bf3ee642)

@AbandonedCart It would be great if you could finish this pull request, if you want to

**Comment by AbandonedCart** [comment]
Sure. I can do that.

**Comment by dima-xd** [comment]
Closing because commit moved to https://github.com/skyline-emu/skyline/pull/2312

**Pull request closed without merging**